odl/contrib/fom/supervised.py

```diff
@@ -790,16 +853,26 @@ def noise_power_spectrum(data, ground_truth, radial=False,
     Returns
     -------
     noise_power_spectrum : `DiscreteLp`-element
-        The space is the Fourier space corresponding to ``data.space``, and
+        The space is the Fourier space corresponding to ``space``, and
         hence the axes indicate frequency.
         If ``radial`` is ``True``, an average over concentric annuli is
         taken. The result is an element of a one-dimensional space with
         domain ``[0, rmax]``, where ``rmax`` is the radius of the smallest
-        ball containing ``data.space.domain``. Its shape is ``(N,)`` with ::
+        ball containing ``space.domain``. Its shape is ``(N,)`` with ::
 
             N = int(sqrt(sum(n ** 2 for n in image.shape)) / binning_factor)
     """
-    ft = odl.trafos.FourierTransform(data.space, halfcomplex=False)
+    try:
+        space = data.space
+        assert isinstance(space, odl.DiscreteLp)
```

Comment:
Meeeh

Reply:
C'mon, it's `contrib` :-)

Reply:
Actually no, it's caught in the next line and handled. So no meh